cmd/sink/filters/filters.go

```diff
@@ -128,34 +127,83 @@ func (f *Filters) createFilters(
 	if err != nil {
 		return archiveMap, deleteMap, archiveOnDeleteMap, err
 	}
+
 	var errList []error
 	for _, resource := range resources {
 		gvk := schema.FromAPIVersionAndKind(resource.Selector.APIVersion, resource.Selector.Kind)
 		namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, gvk)
 		if resource.ArchiveWhen != "" {
-			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk.String()])
+			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				archiveMap[namespaceGvk] = *archiveExpr
 			}
 		}
 		if resource.DeleteWhen != "" {
-			deleteExpr, err := ocel.CreateCelExprOr(resource.DeleteWhen, globalDelete[gvk.String()])
+			deleteExpr, err := ocel.CreateCelExprOr(resource.DeleteWhen, globalDelete[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				deleteMap[namespaceGvk] = *deleteExpr
 			}
 		}
 		if resource.ArchiveOnDelete != "" {
-			archiveOnDeleteExpr, err := ocel.CreateCelExprOr(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk.String()])
+			archiveOnDeleteExpr, err := ocel.CreateCelExprOr(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				archiveOnDeleteMap[namespaceGvk] = *archiveOnDeleteExpr
 			}
 		}
+
+		// Now we cycle the global filters `globalArchive`, `globalDelete` and `globalArchiveOnDelete`
+		// compiling and adding any global filter that was not added already. Here the resources that
+		// are globally filtered but not locally filtered are added.
+		for globalGvk, expression := range globalArchive {
+			namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, globalGvk)
+			_, exists := archiveMap[namespaceGvk]
+			if exists {
+				continue
+			}
+
+			expressionCEL, err := ocel.CreateCelExprOr(expression)
```

Comment:
Can and should we rename `CreateCelExprOr`? 

What this method does is "or" the given CEL expressions together and return the compiled CEL program.  How about `CompileOrdCELExpression?  It really only does the "or" and then calls `CompileCELExpression`.

And can we standardize on "Cel" or CEL"?  From what I see in the documentation, they use  "CEL".

Reply:
I changed it.